types/chromecast-caf-receiver/cast.framework.ui.d.ts

```diff
@@ -1,197 +1,193 @@
-
-        /**
-         * Content Type.
-         */
-        contentType: ContentType;
-
-        /**
-         * The number of the current playing break clip in the break.
-         */
-        currentBreakClipNumber: number;
-
-        /**
-         * Media current position in seconds; or break current position if playing break.
-         */
-        currentTime: number;
-
-        /**
-         * Whether the player metadata (ie: title; currentTime) should be displayed.
-         *  This will be true if at least one field in the metadata should be displayed.
-         *  In some cases; displayStatus will be true; but parts of the metadata should be hidden
-         * (ie: the media title while media is seeking).
-         * In these cases; additional css can be applied to hide those elements.
-         * For cases where the media is audio-only; this will almost always be true.
-         * In cases where the media is video; this will be true when:
-         *   (1) the video is loading; buffering; or seeking
-         *   (2) a play request was made in the last five seconds while media is already playing;
-         *   (3) there is a request made to show the status in the last five seconds; or
-         *   (4) the media was paused in the last five seconds.
-         */
-        displayStatus: boolean;
-
-        /**
-         * Media duration in seconds; Or break duration if playing break.
-         */
-        duration: number;
-
-        /**
-         * Indicate break clip can be skipped.
-         */
-        isBreakSkippable: boolean;
-
-        /**
-         * Indicate if the content is a live stream.
-         */
-        isLive: boolean;
-
-        /**
-         * Indicate that the receiver is playing a break.
-         */
-        isPlayingBreak: boolean;
-
-        /**
-         * Indicate the player is seeking (can be either during playing or pausing).
-         */
-        isSeeking: boolean;
-
-        /**
-         * Media metadata.
-         */
-        metadata: MediaMetadata;
-
-        /**
-         * Next Item subtitle.
-         */
-        nextSubtitle: string;
-
-        /**
-         * Next Item thumbnail url.
-         */
-        nextThumbnailUrl: string;
-
-        /**
-         * Next Item title.
-         */
-        nextTitle: string;
-
-        /**
-         * Number of break clips in current break.
-         */
-        numberBreakClips: number;
-
-        /**
-         * Flag to show/hide next item metadata.
-         */
-        preloadingNext: boolean;
-
-        /**
-         * Current player state.
-         */
-        state: State;
-
-        /**
-         * Content thumbnail url.
-         */
-        thumbnailUrl: string;
-
-        /**
-         * Content title.
-         */
-        title: string;
-
-        /**
-         * Provide the time a break is skipable - relative to current playback time. Undefined if not skippable.
-         */
-        whenSkippable?: number;
-    }
+
+export as namespace ui
```

Comment:
It looks like this will make it possible to reference these types as `ui.Type` as well as `cast.framework.ui.Type`.  Is that intentional?

Reply:
So, to take `PlayerData` in cast.framework.ui.d.ts as an example, the goal is to be able to either `import { PlayerData } from "chromecast-caf-receiver/cast.framework.ui"` in a module or refer to it as `cast.framework.ui.PlayerData` in a global context?

Reply:
yes, exactly

Reply:
Some readers may find the current solution a little hard to read, but it sounds like you've found a way to satisfy all of your goals.  My only suggestion would be to add a little more test coverage to indicate the different kinds of references you expect to work (in particular, imports).

Reply:
Ok @amcasey - I have added an example with some comments to the tests. I had to ignore tslint rules for using `cast.framework` usage in the tests. Thanks for your help today :)

Reply:
Thanks!